09_json/willshen8/pkg/puppy/json/jsonifyPuppy.go

```diff
@@ -0,0 +1,22 @@
+package puppy
+
+import (
+	"encoding/json"
+
+	puppy "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy"
+	store "github.com/anz-bank/go-course/09_json/willshen8/pkg/puppy/store"
+)
+
+func UnmarshalPuppy(jsonData []byte, decodedPuppies *[]puppy.Puppy) error {
+	if unMarshalError := json.Unmarshal(jsonData, &decodedPuppies); unMarshalError != nil {
+		return unMarshalError
+	}
+	return nil
```

Comment:
you might as well write this as `return json.Unmarshal(jsonData, &decodedPuppies)` - which makes the whole function superfluous.

Reply:
Agreed, I started with more code but the refactoring reduced this function to a superfluous one :)

Reply:
I have now removed the whole file and just have it in other functions within `main.go`